Bluetooth: Host: Move BR/EDR files to host/classic

lylezhu2012 opened the pull request:
Add subfolder "subsys/bluetooth/host/classic". Move BR/EDR source files to the subfolder.

Move BR/EDR header files from "include/zephyr/bluetooth" to subfolder "include/zephyr/bluetooth/classic".

Move source files of BR/EDR to subfolder "subsys/bluetooth/host/classic/".

Add the path "include/zephyr/bluetooth/classic/" for the header files of BR/EDR.

lylezhu2012 committed:
Bluetooth: Host: Move BR/EDR files to host/classic

Add subfolder "subsys/bluetooth/host/classic".
Move BR/EDR source files to the subfolder.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 committed:
Bluetooth: Host: Include full path of l2cap_br_interface.h

The header file l2cap_interface.h has been included by
tests/bsim/bluetooth/host/* cases.

To avoid potential issue, include full path of header
l2cap_br_interface.h in file l2cap_interface.h.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 committed:
Bluetooth: Host: Move BR headers to subfolder classic

Move BR/EDR header files from "include/zephyr/
bluetooth" to subfolder "include/zephyr/bluetooth/
classic".

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 committed:
MAINTAINERS: Update Bluetooth Classic file path

Move source files of BR/EDR to subfolder "subsys/
bluetooth/host/classic/".

Add the path "include/zephyr/bluetooth/classic/"
for the header files of BR/EDR.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

Thalley commented on subsys/bluetooth/host/CMakeLists.txt:
```diff
@@ -3,32 +3,16 @@
 zephyr_library()
 zephyr_library_link_libraries(subsys__bluetooth)
 
+add_subdirectory_ifdef(CONFIG_BT_BREDR classic)
```
An open question: 
Should we use classic or BREDR everywhere?
Right now, we have the Kconfigs called BREDR, but the directory is called classic. 

So the new directory be `bredr` instead? Or should the Kconfig be renamed to `CONFIG_BT_CLASSIC`? 

That way we would be consistent with BT Mesh and BT Audio

jori-nordic replied:
I would also prefer `BT_CLASSIC`.
We can keep the `BREDR` kconfig as deprecated and make it select `CLASSIC`.

Thalley commented on subsys/bluetooth/host/Kconfig:
```diff
@@ -1028,6 +1028,7 @@ config BT_CONN_DISABLE_SECURITY
 config BT_BREDR
 	bool "Bluetooth BR/EDR support [EXPERIMENTAL]"
 	depends on BT_HCI_HOST
+	select BT_CLASSIC
```
Given that `BT_BREDR` is considered experimental, can't we just replace the name of the Kconfig? AFAIK we don't do deprecation of experimental things (even if it has been experimental for a long time...) 

If we do a simple rename, then we should put that in the migration guide for 3.7

lylezhu2012 replied:
It is done. 

Could you guide me how to update the migration guide for 3.7?

Thalley replied:
@lylezhu2012 It's basically just adding an entry in https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/migration-guide-3.7.rst (maybe under a new section called Bluetooth Classic next to the Mesh and Audio sections) 

You can see the previous migrations guides at https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/ to get an idea of what to write :)

lylezhu2012 replied:
Got it. Thanks a lot.

lylezhu2012 committed:
Bluetooth: Rename BT_BREDR to BT_CLASSIC

Rename BT_BREDR to BT_CLASSIC
Rename CONFIG_BT_BREDR to CONFIG_BT_CLASSIC

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

Thalley reviewed (changes requested):
Missing migration guide w.r.t the Kconfig rename

lylezhu2012 committed:
doc: releases: migration-guide-3.7: Rename BT_BREDR to BT_CLASSIC

Due to the source/header files are moved to subfolder "classic",
rename the kconfig option BT_BREDR to BT_CLASSIC to make the
folder name and kconfig option name consistent.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 commented:
> Missing migration guide w.r.t the Kconfig rename

lylezhu2012 commented:
> Missing migration guide w.r.t the Kconfig rename

Updated migration guide 3.7

jhedberg commented:
@lylezhu2012 there seems to be some conflicts that need resolving.
